dash/dash.py

```diff
@@ -1178,28 +1186,75 @@ def clientside_callback(
         ```
 
         With this signature, Dash's front-end will call
-        `window.my_clientside_library.my_function` with the current
-        values of the `value` properties of the components
-        `my-input` and `another-input` whenever those values change.
-
-        Include a JavaScript file by including it your `assets/` folder.
-        The file can be named anything but you'll need to assign the
-        function's namespace to the `window`. For example, this file might
-        look like:
+        `window.dash_clientside.my_clientside_library.my_function` with the
+        current values of the `value` properties of the components `my-input`
+        and `another-input` whenever those values change.
+
+        Include a JavaScript file by including it your `assets/` folder. The
+        file can be named anything but you'll need to assign the function's
+        namespace to the `window.dash_clientside` namespace. For example,
+        this file might look like:
+        ```
+        window.dash_clientside = Object.assign(window.dash_clientside || {},
+            {
+                my_function: function(input_value_1, input_value_2) {
+                    return (
+                        parseFloat(input_value_1, 10) +
+                        parseFloat(input_value_2, 10)
+                    );
+                }
+            }
+        );
         ```
-        window.my_clientside_library = {
-            my_function: function(input_value_1, input_value_2) {
+
+        Alternatively, you can pass the JavaScript source directly to
+        `clientside_callback`. In this case, the same example would look like:
+        ```
+        app.clientside_callback(
+            '''
+            function(input_value_1, input_value_2) {
                 return (
                     parseFloat(input_value_1, 10) +
                     parseFloat(input_value_2, 10)
                 );
             }
-        }
+            ''',
+            Output('my-div' 'children'),
+            [Input('my-input', 'value'),
+             Input('another-input', 'value')]
+        )
         ```
         """
         self._validate_callback(output, inputs, state)
         callback_id = _create_callback_id(output)
 
+        # If JS source is explicitly given, create a namespace and function
+        # name, then inject the code.
+        if isinstance(clientside_function, str):
+
+            out0 = output
+            if isinstance(output, (list, tuple)):
+                out0 = output[0]
+
+            namespace = '_dashprivate_{}'.format(out0.component_id)
+            function_name = '{}'.format(out0.component_property)
+
+            self._inline_scripts.append(
+                """
+                if (!window.dash_clientside) {{
+                    window.dash_clientside = {{}};
+                }}
+                window.dash_clientside["{0}"] = Object.assign(
```

Comment:
Oh hmm, not sure this is safe since we still support IE11. Looks like right now it works as our current bundle pipeline includes a polyfill for `Object.assign`, but especially with all the performance work @Marc-Andre-Rivet is doing I'm not sure I trust us to always do it that way, could potentially be converted to transpiling and this would break.

Also while we're on the topic of pathological ids and props (good call putting the id in the namespace!), I think `"` is technically allowed in either one, so should be escaped.

Depending on how golf-like you want to be, could be something like
```
var clientside = window.dash_clientside = window.dash_clientside || {{}};
var ns = clientside["{0}"] = clientside["{0}"] || {{}};
ns["{1}"] = {2};
```

Reply:
works for me!